roles/openstack-package/tasks/main.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: add openstack package repo key
+  apt_key: url='{{ openstack_package.repo_key_url }}'
+
+- name: add openstack package repo
+  apt_repository: repo='{{ openstack_package.repo }}'
+
+- name: "install {{ project_name }} package"
+  apt: pkg="{{ package_name|default(openstack_package.package_name) }}" 
+       update_cache=yes cache_valid_time=3600
+
+- name: setup openstack current base path
+  file: dest=/opt/openstack state=directory
+
+- name: test whether current {{ project_name }} symlink exists
+  shell: 'test -h /opt/openstack/current'
+  register: symlink_exists
+  ignore_errors: True
```

Comment:
Why not use the stat module here? You don't have to ignore errors, and the stat details will tell you if it is a symlink or if it exists or not.